counter-sink/src/main/java/org/springframework/cloud/stream/module/metrics/CounterSink.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.metrics;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.metrics.CounterService;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableModule;
+import org.springframework.cloud.stream.annotation.Sink;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.context.IntegrationContextUtils;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.util.Assert;
+
+/**
+ * A simple module that counts messages received, using Spring Boot metrics abstraction.
+ *
+ * @author Eric Bottard
+ */
+@EnableModule(Sink.class)
+@EnableConfigurationProperties(CounterSinkOptions.class)
+public class CounterSink {
+
+	private static Logger logger = LoggerFactory.getLogger(CounterSink.class);
+
+	private CounterService counterService;
+
+	private CounterSinkOptions options;
+
+	// Default value useful for tests
+	private EvaluationContext evaluationContext = new StandardEvaluationContext();
+
+	private boolean evaluationContextSet;
+
+	private BeanFactory beanFactory;
+
+	@Autowired
+	public void setOptions(CounterSinkOptions options) {
+		this.options = options;
+	}
+
+	@Autowired
+	public void setCounterService(CounterService counterService) {
+		this.counterService = counterService;
+	}
+
+	@ServiceActivator(inputChannel=Sink.INPUT)
+	public void counterSink(Message<?> message) {
```

Comment:
That one was in place since we only had 2 modules "time" and "log". I agree we should change it. This is something we need to determine in general - what is the convention or do we even need to supply a name here vs generating a default (e.g. application.name)? We do need to keep in mind that a spring-cloud-stream module should eventually be runnable as a self-contained app with non-spring-cloud-stream app(s) interacting with it from the other side of the broker.